## Домашнє завдання 6: Статистичні тести

0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [48]:
import pandas as pd
import numpy as np
from statsmodels.stats.weightstats import ztest

data_path = 'data/data.csv'
df = pd.read_csv(data_path)
data = df["data"].to_numpy()

Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [72]:
data.shape

(100,)

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

In [100]:
null_mean = 20

ztest_Score, p_value = ztest(data, value = null_mean, alternative='larger')
print(ztest_Score, p_value)

def statistical_significance_test(p_value: int):
    alpha = 0.05
    if(p_value <  alpha):
        print("Відхилити Н0.")
    else:
        print("Н0 не може бути відхилена.")

statistical_significance_test(p_value)

-0.9953477389335053 0.8402164165464012
Н0 не може бути відхилена.


Оскільки p-value > 0.05, нульова гіпотеза не може бути відхилена. Іншими словами, у нас немає доказів, що середній чек статистично значуще збільшився.

2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [92]:
data_path_2 = 'data/data_2.csv'
df_2 = pd.read_csv(data_path_2)
data_2 = df_2["data"].to_numpy()

In [94]:
np.mean(data), np.mean(data_2)

(19.37755319670441, 19.37322276512519)

In [108]:
ztest_Score, p_value = ztest(data, data_2, alternative='two-sided')
print(ztest_Score, p_value)

statistical_significance_test(p_value)

0.005246673456808504 0.9958137794593356
Н0 не може бути відхилена.


Статистика для двох вибірок z-критерію становить 0.0052, а відповідне p-value дорівнює 0.9958.

Оскільки p-value > 0.05, нульова гіпотеза не може бути відхилена. Іншими словами, статистично значущих змін у середньому чеку між двома місяцями немає.